src/neighbors.cc

```diff
@@ -732,7 +732,7 @@ peerDigestLookup(CachePeer * p, PeerSelector * ps)
     assert(p->digest->cd);
     /* does digest predict a hit? */
 
-    if (!p->digest->cd->contains(key))
+    if (key && !p->digest->cd->contains(key))
```

Comment:
> Found by static analysis tools

Can those tools prove that key may be nil here? If not, then I hope that humans can and, hence, should, assert that key is never nil in this context. If you agree, please refactor the code in the beginning of peerDigestLookup() accordingly:

```C++
    assert(ps);
    HttpRequest *request = ps->request;
    assert(request);

    assert(p);
```

And then just do this here (instead of dangerously storing ephemeral storeKeyPublicByRequest() result as if for reuse):

```suggestion
    if (!p->digest->cd->contains(storeKeyPublicByRequest(request)))
```

We could `assert(key)` above, but I hope that static analysis will discover that key cannot be nil in this code variation without that misleading noise.

We should document that storeKeyPublicByRequest() never returns nil, but it may be better to spend that energy on changing cache_key type to a non-pointer instead.

Reply:
Got it. Sorry, for shallow analysis, will be more attentive next time. 
I will take into account your suggestions and add required changes

Reply:
No problem at all! Looking forward to your changes.